lib/pushnotificationsservice.js

```diff
@@ -0,0 +1,350 @@
+    filename: 'txp_finally_rejected',
+  }
+};
+
+function PushNotificationsService() {};
+
+PushNotificationsService.prototype.start = function(opts, cb) {
+
+  opts = opts || {};
+
+  function _readDirectories(basePath, cb) {
+    fs.readdir(basePath, function(err, files) {
+      if (err) return cb(err);
+      async.filter(files, function(file, next) {
+        fs.stat(path.join(basePath, file), function(err, stats) {
+          return next(!err && stats.isDirectory());
+        });
+      }, function(dirs) {
+        return cb(null, dirs);
+      });
+    });
+  };
+
+  var self = this;
+  self.templatePath = path.normalize(((__dirname + '/templates')) + '/');
+  self.defaultLanguage = 'en';
+  self.defaultUnit = 'btc';
+  self.subjectPrefix = '';
+  self.publicTxUrlTemplate = {};
+
+  async.parallel([
+    function(done) {
+      _readDirectories(self.templatePath, function(err, res) {
+        self.availableLanguages = res;
+        done(err);
+      });
+    },
+    function(done) {
+      self.storage = new Storage();
+      self.storage.connect(opts.storageOpts, done);
+    },
+    function(done) {
+      self.messageBroker = opts.messageBroker || new MessageBroker(opts.messageBrokerOpts);
+      self.messageBroker.onMessage(_.bind(self._sendPushNotifications, self));
+      done();
+    },
+  ], function(err) {
+    if (err) {
+      log.error(err);
+    }
+    return cb(err);
+  });
+};
+
+PushNotificationsService.prototype._sendPushNotifications = function(notification, cb) {
+  var self = this;
+  var url = 'http://192.168.1.143:8000/send';
+  cb = cb || function() {};
+
+  var notifType = PUSHNOTIFICATIONS_TYPES[notification.type];
+  if (!notifType) return cb();
+
+  self._getRecipientsList(notification.walletId, function(err, recipientsList) {
+    self.storage.fetchWallet(notification.walletId, function(err, wallet) {
+
+      var resultedRecipientsList = _.reject(self._getJoinedRecipientsList(wallet, recipientsList), {
+        id: notification.creatorId || null
+      });
+      async.waterfall([
+        function(next) {
+          self._readAndApplyTemplates(notification, notifType, resultedRecipientsList, next);
+        },
+        function(contents, next) {
+          async.map(resultedRecipientsList, function(recipient, next) {
+            var opts = {};
+            var content = contents[recipient.language];
+            opts.users = [notification.walletId + '$' + recipient.id];
+            opts.android = {
+              "data": {
+                "title": content.plain.subject,
+                "message": content.plain.body
+              }
+            };
+            opts.ios = {
+              "badge": 0,
+              "alert": content.plain.body,
+              "sound": "soundName"
+            };
+            return next(err, opts);
+          }, next);
+        },
+        function(optsList, next) {
+          async.each(optsList,
+            function(opts, next) {
+              request({
```

Comment:
It would be nice to have a `PushNotificationService.prototype._makeRequest` function to improve code readability. Also including some logs when notifications fail would be great.